### PR TITLE
Test case for parsing static dns when device is present on system

### PR DESCRIPTION
**Pull request by yawkat**
This patch adds a test case for the launchpad bug below. When parsing a network configuration where a device on the system has the same mac address as given in the `match` section of the config, but with the real device name not matching the config name, parsing of a static nameserver entry fails with a `KeyError`.

LP: #1979877

### CODE REVIEW

**Comment by yawkat** [comment]
Note on CLA: I cannot sign the individual contributor CLA because it requires approval from my employer. However I've been told in the launchpad bug discussion that my employer (Oracle) has an existing agreement that allows employees to contribute to this project.

**Comment by blackboxsw** [comment]
> Note on CLA: I cannot sign the individual contributor CLA because it requires approval from my employer. However I've been told in the launchpad bug discussion that my employer (Oracle) has an existing agreement that allows employees to contribute to this project.

Thanks @yawkat. I've confirmed as well that CLA between Canonical/Oracle is in place and I have sent you an email from launchpad to the email address you have in Launchpad to confirm your affiliation w/ Oracle to this github account so we can attribute PRs from https://github.com/yawkat as approved and no individual CLA signature would be needed in this case.

**Comment by holmanb** [comment]
@yawkat: The ["fix" has been merged](https://github.com/canonical/cloud-init/pull/1552). I think this PR needs a couple minor tweaks before merging.

- format change from the GH actions CI: tests/unittests/net/test_dns.py:23:17: W291 trailing whitespace
- The merged fix adds another call to `get_interfaces_by_mac()` that will need to be mocked as well in this test. There is a [patch in the PR](https://github.com/canonical/cloud-init/pull/1552) which I adapted from your original test on Launchpad if that helps.

**Comment by yawkat** [comment]
@holmanb thanks! rebased, the test passes now. i've also adjusted the mocking like in your patch.

**Comment by holmanb** [comment]
@yawkat Thanks! It looks like `black` is complaining about a minor formatting issue still. Running `tox -e do_format` should auto-format it, unless you prefer to do it manually (I think the function parameters are indented too far).

**Comment by blackboxsw** [comment]
> Thanks @yawkat. I've confirmed as well that CLA between Canonical/Oracle is in place and I have sent you an email from launchpad to the email address you have in Launchpad to confirm your affiliation w/ Oracle to this github account so we can attribute PRs from https://github.com/yawkat as approved and no individual CLA signature would be needed in this case.

Confirmed email receipt from appropriate oracle.com authenticated adddress. CLA is good

**Review by blackboxsw — APPROVED** [review]
Approved from me, I'll let @holmanb drive this to review/merge.

**Review by holmanb — APPROVED** [review]
Thanks for your help on this @yawkat!

**Pull request merged**
